src/base/static/state/ducks/map.js

```diff
@@ -1,311 +1,581 @@
-// Selectors
-export const mapBasemapSelector = state => {
-  return state.map.visibleBasemapId;
-};
-export const mapLayerStatusesSelector = state => {
-  return state.map.layerStatuses;
-};
-export const mapSizeValiditySelector = state => {
-  return state.map.isMapSizeValid;
-};
-export const mapPositionSelector = state => {
-  return state.map.mapPosition;
-};
-export const mapDraggedSelector = state => {
-  return state.map.isMapDragged;
-};
-export const mapboxStyleIdSelector = state => {
-  return (
-    Object.values(state.map.layerStatuses).find(layerStatus => {
-      return layerStatus.type === "mapbox-style" && layerStatus.isVisible;
-    }) || {}
-  ).id;
-};
-export const mapFeatureFiltersSelector = state => {
-  return state.map.featureFilters;
-};
-export const mapUpdatingFilterGroupIdSelector = state => {
-  return state.map.updatingFilterGroupId;
-};
-export const mapUpdatingFilterTargetLayerSelector = state => {
-  return state.map.updatingFilterTargetLayer;
-};
-export const mapDraggingSelector = state => {
-  return state.map.isMapDragging;
-};
+import PropTypes from "prop-types";
+import { FlyToInterpolator } from "react-map-gl";
 
-// Actions
-const SET_LAYER_LOAD_STATUS = "map/SET_LAYER_LOAD_STATUS";
-const SET_MAP_POSITION = "map/SET_MAP_POSITION";
-const SET_MAP_SIZE_VALIDITY = "map/SET_MAP_SIZE_VALIDITY";
-const ACTIVATE_FEATURE_FILTER = "map/ACTIVATE_FEATURE_FILTER";
-const DEACTIVATE_FEATURE_FILTER = "map/DEACTIVATE_FEATURE_FILTER";
-const RESET_FEATURE_FILTER_GROUP = "map/RESET_FEATURE_FILTER_GROUP";
-const SET_BASEMAP = "map/SET_BASEMAP";
-const INIT_LAYER = "map/INIT_LAYER";
-const UPDATE_LAYERS_VISIBLE = "map/UPDATE_LAYERS_VISIBLE";
-const UPDATE_LAYERS_HIDDEN = "map/UPDATE_LAYERS_HIDDEN";
-const UPDATE_MAP_DRAGGING = "map/UPDATE_MAP_DRAGGING";
-const UPDATE_IS_MAP_DRAGGED = "map/UPDATE_IS_MAP_DRAGGED";
+import { easeInOutCubic } from "../../utils/scroll-helpers";
```

Comment:
We borrow the easing function from scroll helpers, which might be confusing but it's the same function we would use anyway.